poppy/accel_math.py

```diff
@@ -34,19 +31,46 @@
     ne = None
     _NUMEXPR_AVAILABLE = False
 
-_USE_CUDA = (conf.use_cuda and _ACCELERATE_AVAILABLE)
+try:
+    # try to import anaconda accelerate package to see if it is available
+    import pyculib
+    from numba import cuda
+    _CUDA_PLANS = {} # plans for various array sizes already prepared
+    _CUDA_AVAILABLE = True
+except ImportError:
+    pyculib = None
+    _CUDA_AVAILABLE = False
+
+try:
+    # try to import pyopencl and gpyfft to see if OpenCL FFT is available
+    import pyopencl
+    import pyopencl.array
+    import gpyfft
+    _OPENCL_AVAILABLE = True
+    _OPENCL_STATE = dict()
+except ImportError:
+    _OPENCL_AVAILABLE = False
+
+
+_USE_CUDA = (conf.use_cuda and _CUDA_AVAILABLE)
+_USE_OPENCL = (conf.use_numexpr and _OPENCL_AVAILABLE)
 _USE_NUMEXPR = (conf.use_numexpr and _NUMEXPR_AVAILABLE)
```

Comment:
shouldn't there by a _USE_FFTW for uniformity?

Reply:
also, what if _USE_CUDA and _USE_OPENCL are both True, is that handled somewhere else?

Reply:
It's handled implicitly by the order of the if/elif inside `fit_2d`.  _USE_CUDA has precedence.

Reply:
at the moment, I'm flipping these bools to run my benchmarks, that's why I'm paying particular attention to them.

Reply:
Right. I was trying to get it so that one could flip the `poppy.conf.use_x` bools and it would have the desired effect. On the general principle that what you're trying to do is configure a setting, so it should be possible via the configuration settings system. :-)  

But I see that doesn't quite work right for all the cases of interest right now, since the file global versions are still accessed in a bunch of places.

Reply:
well, it only costs a couple microseconds to just evaluate `(conf.use_cuda and _CUDA_AVAILABLE)` on the fly when needed. So there is very small cost to putting that check directly in the fftshift function and so on. That would be one way to let the `conf` settings be toggled at runtime and work.

Reply:
It's all a bit confusing but that seems like a reasonable path. I don't think the config file is as critical as having a setting that they can adjust at run time in their own script.